lib/constants.js

```diff
@@ -9,6 +9,9 @@
  */
 
 module.exports = {
+  // The default span or transaction `type`.
+  DEFAULT_SPAN_TYPE: 'custom',
```

Comment:
REVIEW NOTE: The spec says the default span.type should be 'custom'. We had been doing that in Span serialization. However, when inferring `span.context.service.target.type` in `Span.end()`, the algorithm needs that `span.type` value. One way was to have a shared constant -- which I've done here. Another way would be to actually set `span.type = 'custom'` as a default.